index.d.ts

```diff
@@ -626,3 +626,51 @@ interface KVNamespace {
     cursor: string;
   }>;
 }
+
+interface DurableObjectEntries<T> {
+  [key: string]: T;
+}
+
+interface DurableObjectListOptions {
+  start: string;
+  end: string;
+  reverse: boolean;
+  limit: number;
+}
+
+interface DurableObjectStorage {
+  get<T = unknown>(key: string): Promise<T>;
+  get<T = unknown>(keys: Array<string>): Promise<Map<string, T>>;
+  put<T = unknown>(key: string, value: T): Promise<void>;
+  put<T = unknown>(entries: DurableObjectEntries<T>): Promise<void>;
+  delete(key: string): Promise<boolean>;
+  delete(keys: Array<string>): Promise<boolean>;
+  list<T = unknown>(options?: DurableObjectListOptions): Promise<Map<string, T>>;
+  transaction<T = unknown>(
+    closure: (txn: DurableObjectStorage) => Promise<void>
```

Comment:
As it stands, typing txn as a DurableObjectStorage implies that nested transactions are possible, which they do not seem to be (I tested and got "TypeError: tx.transaction is not a function").  

The txn object probably needs its own type. If you want to separate it out into the pieces of the interface that only the txn has vs the pieces the storage object has, should be able to avoid duplication.

Either way, besides not having a transaction method, the txn object also needs a rollback method on it.